// [[Reason: Aliases are preserved for binding patterns GH#55654]] ////

//// [tests/cases/compiler/declarationEmitKeywordDestructuring.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -22,23 +22,23 @@
     async: boolean;
     await: boolean;
     one: boolean;
 };
-declare function f3({ abstract, ...rest }: P): {
+declare function f3({ abstract: _abstract, ...rest }: P): {
     enum: boolean;
     function: boolean;
     async: boolean;
     await: boolean;
     one: boolean;
 };
-declare function f4({ async, ...rest }: P): {
+declare function f4({ async: _async, ...rest }: P): {
     enum: boolean;
     function: boolean;
     abstract: boolean;
     await: boolean;
     one: boolean;
 };
-declare function f5({ await, ...rest }: P): {
+declare function f5({ await: _await, ...rest }: P): {
     enum: boolean;
     function: boolean;
     abstract: boolean;
     async: boolean;
